pkg/endpoint/bpf.go

```diff
@@ -1053,7 +1074,7 @@ type policyMapPressureUpdater interface {
 }
 
 func (e *Endpoint) updatePolicyMapPressureMetric() {
-	value := float64(e.realizedPolicy.GetPolicyMap().Len()) / float64(e.policyMap.MaxEntries())
+	value := float64(e.desiredPolicy.GetPolicyMap().Len()) / float64(e.policyMap.MaxEntries())
```

Comment:
I think this is a separate change that we should separately review, right?

Reply:
The reason it is important here is that `realizedPolicy` is going to be brought down to ~6 entries, so the metric will be wrong during lockdown mode.

Reply:
Does that mean we will now still report a policymap overflow via the metric even though technically the capacity of the policymap has completely changed? I feel like we might be better off with a new metric that provides visibility into the lockdown state of the map as well, rather than co-opting this BPF map pressure metric and repurposing it in this special case.

Reply:
The absolute capacity has changed in lockdown mode, but the effective capacity should be measured against what the capacity would be usually.

This change maintains how the metric works now. In main `realizedPolicy` will blindly accept new entries while bpf insertion fails, so `bpf_map_pressure` is expected to go above `1.0`.

If we don't change how this metric is calculated, the metric will falsely report a ratio of 6/16384 `bpf_map_pressure` in lockdown mode, which is technically correct but effectively incorrect.

Adding a new metric for lockdown might be a useful idea, but `bpf_map_pressure` already conveys the same information at values `>=1.0`.

Reply:
> realizedPolicy and desiredPolicy are the same map

They can be, but during regeneration (and now lockdown) they are not.